packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -14,7 +14,7 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
   private _allowInsecure = false;
   private readTimeout = 1000;
 
-  constructor(private readonly path: string, private readonly port: number) {
+  constructor(private readonly path: string, private readonly port: number, private readonly shutdownOnExit = false) {
```

Comment:
added flag for backward compatibility because I don't know how it will work with retry start option for container

if it 'true' it will throw error and stop whole process on container shutdown